test/helpers/record_helper_test.rb

```diff
@@ -101,4 +101,149 @@ class RecordHelperTest < ActionView::TestCase
+                        'rights' => [
+                          {
+                            'description' => 'foo',
+                            'kind' => 'bar'
+                          },
+                          {
+                            'description' => 'Free/open to all',
+                            'kind' => 'Access to files'
+                          }
+                        ]
+                      }
+    assert_nil gis_access_link(rights_no_links)
+  end
+
+  test 'gis_access_link is website URL for non-MIT records' do
+    access_elsewhere = {
+                         'rights' => [
+                           {
+                             'description' => 'foo',
+                             'kind' => 'bar'
+                           },
+                           {
+                             'description' => 'Not owned by MIT',
+                             'kind' => 'Access to files'
+                           }
+                         ],
+                         'links' => [
+                           {
+                             'kind' => 'Download',
+                             'text' => 'Data',
+                             'url' => 'https://example.org/dz_f7regions_2016.zip'
+                           },
+                           {
+                             'kind' => 'Website',
+                             'text' => 'Website',
+                             'url' => 'https://example.org/gismit:dz_f7regions_2016'
+                           }
+                         ],
+                         'provider' => 'Spelman'
+                       }
+    assert_equal 'https://example.org/gismit:dz_f7regions_2016', gis_access_link(access_elsewhere)
+  end
+
+  test 'gis_access_link is data download URL for MIT records' do
+    access_free = {
+                    'rights' => [
+                      {
+                        'description' => 'foo',
+                        'kind' => 'bar'
+                      },
+                      {
+                        'description' => 'Free/open to all',
+                        'kind' => 'Access to files'
+                      }
+                    ],
+                    'links' => [
+                      {
+                        'kind' => 'Download',
+                        'text' => 'Data',
+                        'url' => 'https://example.org/dz_f7regions_2016.zip'
+                      },
+                      {
+                        'kind' => 'Website',
+                        'text' => 'Website',
+                        'url' => 'https://example.org/gismit:dz_f7regions_2016'
+                      }
+                    ]
+                  }
+    access_auth = {
+                    'rights' => [
+                      {
+                        'description' => 'foo',
+                        'kind' => 'bar'
+                      },
+                      {
+                        'description' => 'Restricted',
```

Comment:
This may be too finicky a comment, in which case I apologize for being pedantic. I'm trying to make sure that I understand the logic of the helper method, and I'm not sure whether a record like this will actually end up generating an authenticated button, or an external link.

The check in the access_button partial appears to be looking for the access_type helper to return `MIT Authentication` as a value, rather than `Restricted`?

I also realize that this test is only testing the `gis_access_link` helper, and that we aren't trying to test the view results. So maybe this is not something that needs to be adjusted. I bring it up only because I think this is what was proving tricky for me to understand as I tried to follow the helper logic.

Reply:
Yes, that's a good point. I think this was an oversight on my part. While the test outcome isn't affected by the access type in this case, it's a good idea to keep them consistent.